src/utils/helpers/validatePage.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  ApplicationElementStates,
+  ElementState,
+  ProgressStatus,
+  ResponsesFullByCode,
+  ResponseFull,
+} from '../../utils/types'
+
+import { TemplateElementCategory } from '../../utils/generated/graphql'
+
+export const PROGRESS_STATUS: { [key: string]: ProgressStatus } = {
+  NOT_VALID: 'NOT_VALID',
+  VALID: 'VALID',
+  INCOMPLETE: 'INCOMPLETE',
+}
+
+export const getCombinedStatus = (
+  array: { status: ProgressStatus }[] | undefined,
+  strictly: boolean = false
+): ProgressStatus => {
+  if (!array) return PROGRESS_STATUS.VALID
+  if (strictly) {
+    return array.every(({ status }) => status === PROGRESS_STATUS.VALID)
+      ? PROGRESS_STATUS.VALID
+      : array.some(({ status }) => status === PROGRESS_STATUS.NOT_VALID)
+      ? PROGRESS_STATUS.NOT_VALID
+      : PROGRESS_STATUS.INCOMPLETE
+  } else {
+    return array.some(({ status }) => status === PROGRESS_STATUS.VALID)
+      ? PROGRESS_STATUS.VALID
+      : array.some(({ status }) => status === PROGRESS_STATUS.NOT_VALID)
+      ? PROGRESS_STATUS.NOT_VALID
+      : PROGRESS_STATUS.INCOMPLETE
+  }
+}
+
+interface validatePageProps {
+  elementsState: ApplicationElementStates
+  responses: ResponsesFullByCode
+  sectionIndex: number
+  page: number
+  checkEmpty?: boolean
+}
+
+const validatePage = ({
+  elementsState,
+  responses,
+  sectionIndex,
+  page,
+  checkEmpty = false,
+}: validatePageProps): ProgressStatus => {
+  let count = 1
+  const elementsInCurrentPage = Object.values(elementsState)
+    .filter(({ section }) => section === sectionIndex)
+    .reduce((pageElements: ElementState[], element) => {
+      if (element.elementTypePluginCode === 'pageBreak') count++
+      if (count !== page) return pageElements
+      if (element.category === TemplateElementCategory.Question) return [...pageElements, element]
```

Comment:
not a required change, this is stylistic preference, but to me logically it makes more sense to check in this fashion:

if pageBreak increment
if not question type category return pageElement
if current page return pageElements + element
return pageElements

Reply:
Thanks. I have changed it slightly to what you have suggested: [here](https://github.com/openmsupply/application-manager-web-app/pull/127/commits/bd0a16c3cc244a73542f8880aa206894619243cc)